prawns/balances/schema.graphql

```diff
@@ -2,110 +2,61 @@ enum SubstrateNetwork {
   kusama
   polkadot
   phala
+  khala
+  litmus
+  litentry
+}
+
+enum SubstrateBalanceChangeEventType {
+  BalancesBalanceSet
+  BalancesDeposit
+  BalancesEndowed
+  BalancesTransfer
+  TreasuryAwarded
 }
 
 type SubstrateBalanceAccount @entity {
-  "address:tokenSymbol"
-  id: ID!
   "address"
-  account: String! @index
+  id: ID!
   "hex address"
-  rootAccount: String! @index
+  publicKey: String! @index
   network: SubstrateNetwork!
   symbol: String!
   decimals: Int!
-  balance: BigInt!
-  firstTransferInDate: DateTime
-  firstTransferInBlockNumber: BigInt
-  firstTransferOutDate: DateTime
-  firstTransferOutBlockNumber: BigInt
-  lastTransferInDate: DateTime
-  lastTransferInBlockNumber: BigInt
-  lastTransferOutDate: DateTime
-  lastTransferOutBlockNumber: BigInt
+  firstBalanceChangeEventDate: DateTime
+  firstBalanceChangeEventBlockNumber: BigInt
+  lastBalanceChangeEventDate: DateTime
+  lastBalanceChangeEventBlockNumber: BigInt
+  totalBalanceChangeEvents: Int!
   totalTransfers: Int!
+  balanceChangeEvents: [SubstrateBalanceChangeEvent]
+    @derivedFrom(field: "account")
   transfersTo: [SubstrateBalanceTransfer] @derivedFrom(field: "to")
   transfersFrom: [SubstrateBalanceTransfer] @derivedFrom(field: "from")
-  treasuryDeposits: [SubstrateTreasuryDeposit] @derivedFrom(field: "depositor")
-}
-
-type SubstrateBalanceTransfer @entity {
-  "network:block:index"
-  id: ID!
-  network: SubstrateNetwork!
-  symbol: String!
-  decimals: Int!
-  toAccountBalanceAtBlock: BigInt!
-  fromAccountBalanceAtBlock: BigInt!
-  to: SubstrateBalanceAccount!
-  from: SubstrateBalanceAccount!
-  amount: BigInt!
-  tip: BigInt!
-  blockNumber: BigInt!
-  date: DateTime!
-}
-
-type SubstrateTreasuryDeposit @entity {
-  "network:block:index"
-  id: ID!
-  depositor: SubstrateBalanceAccount!
-  network: SubstrateNetwork!
-  symbol: String!
-  decimals: Int!
-  accountBalanceAtBlock: BigInt!
-  amount: BigInt!
-  blockNumber: BigInt!
-  date: DateTime!
 }
 
-type SubstrateBalanceDeposit @entity {
+type SubstrateBalanceChangeEvent @entity {
   "network:block:index"
   id: ID!
-  depositor: SubstrateBalanceAccount!
-  depositee: SubstrateBalanceAccount!
   network: SubstrateNetwork!
-  symbol: String!
-  decimals: Int!
-  amount: BigInt!
-  blockNumber: BigInt!
-  date: DateTime!
-}
-
-type SubstrateBalanceEndowed @entity {
-  "network:block:index"
-  id: ID!
   account: SubstrateBalanceAccount!
-  network: SubstrateNetwork!
+  type: SubstrateBalanceChangeEventType!
```

Comment:
Shouldn't we index this though? I guess when we search a certain type of balance change, we might wanna filter by that.

Also, shouldn't we index `account` as well?

Reply:
yeah good spot I missed a few indexes

Reply:
I just ran codegen and spotted that relationships are indexed automatically (makes sense as they are FKs!). Type needed an index though